ui/pages/Project/components/RnaSeqResultPage.jsx

```diff
@@ -0,0 +1,137 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+import { Loader, Grid, Dropdown } from 'semantic-ui-react'
+
+import { getGenesById, getIndividualsByGuid, getRnaSeqDataByIndividual } from 'redux/selectors'
+import { RNASEQ_JUNCTION_PADDING, TISSUE_DISPLAY } from 'shared/utils/constants'
+import DataLoader from 'shared/components/DataLoader'
+import { loadRnaSeqData } from '../reducers'
+import { getRnaSeqDataLoading, getRnaSeqSignificantJunctionData } from '../selectors'
+
+const RnaSeqOutliers = React.lazy(() => import('./RnaSeqOutliers'))
+const RnaSeqOutliersTable = React.lazy(() => import('./RnaSeqJunctionOutliersTable'))
+
+const OUTLIER_VOLCANO_PLOT_CONFIGS = {
+  outliers: {
+    getLocation: (({ geneId }) => geneId),
+    searchType: 'genes',
+    title: 'Expression Outliers',
+    formatData: ({ data }) => data,
+  },
+  spliceOutliers: {
+    getLocation: (({ chrom, start, end }) => `${chrom}:${Math.max(1, start - RNASEQ_JUNCTION_PADDING)}-${end + RNASEQ_JUNCTION_PADDING}`),
+    searchType: 'regions',
+    title: 'Splice Junction Outliers',
+    formatData: ({ data, tissueType }) => data.filter(outlier => outlier.tissueType === tissueType),
+  },
+}
+
+class BaseRnaSeqResultPage extends React.PureComponent {
+
+  static propTypes = {
+    individual: PropTypes.object,
+    rnaSeqData: PropTypes.object,
+    significantJunctionOutliers: PropTypes.arrayOf(PropTypes.object),
+    genesById: PropTypes.object,
+  }
+
+  state = {
+    tissueType: null,
+    tissueOptions: null,
+  }
+
+  componentDidMount() {
+    const { rnaSeqData } = this.props
+    const tissueTypes = Array.from(Object.values(rnaSeqData?.spliceOutliers || {}).flat().reduce(
+      (acc, { tissueType }) => acc.add(tissueType), new Set(),
+    ))
+    const tissueOptions = tissueTypes.map(tissueType => (
+      { key: tissueType, text: TISSUE_DISPLAY[tissueType] || 'No Tissue', value: tissueType }
+    ))
+    if (tissueOptions.length) {
+      this.setState({ tissueType: tissueOptions[0].value, tissueOptions })
+    }
+  }
+
+  onTissueChange = (tissueType) => {
+    this.setState({ tissueType })
+  }
+
+  render() {
+    const { individual, rnaSeqData, significantJunctionOutliers, genesById } = this.props
+    const { tissueType, tissueOptions } = this.state
+
+    const outlierData = Object.entries(rnaSeqData || {}).map(([key, data]) => [
+      key,
+      OUTLIER_VOLCANO_PLOT_CONFIGS[key].formatData({ data: Object.values(data || {}).flat(), tissueType }),
+    ]).filter(([, data]) => data.length)
```

Comment:
I think this would be a bit more stable if instead of iterating over `rnaSeqData` and assuming you have a config for each key, you iterate over a list of configs 
```suggestion
    const outlierPlotConfigs = OUTLIER_VOLCANO_PLOT_CONFIGS.map(({ formatData, ...config }) => ({
      data: formatData(((rnaSeqData || {})[config.key] || {}).flat(), tissueType),
      ...config,
    }).filter(({ data }) => data.length)
```